docs/source/dev/thrift_client.rst

```diff
@@ -0,0 +1,340 @@
+The client is created by using one of two available constructors.
+Which constructor to use depends on how you need to authenticate with the Koverse server.
+There are two ways to authenticate: API Token or user name and password.
+For either constructor, the ``ClientConfiguration`` object must be passed in as well.
+
+Below is an example of authenticating with an API Token:
+
+.. code-block:: java
+
+ import com.koverse.client.thrift.Client;
+ ...
+ String apiToken = ...
+ Client client = new Client(configuration, apiToken);
+ ...
+
+And here is an example of authenticating with a user name and password:
+
+.. code-block:: java
+
+ import com.koverse.client.thrift.Client;
+ ...
+ String userName = ...
+ String password = ...
+ Client client = new Client(configuration, userName, password);
+ ...
+
+After constructing the client, you may use it to perform operations such as:
+
+* Perform queries (JSON object and Lucene based).
+* Get auto-suggest results for a query term.
+* Get a sample of records from a Data Set.
+* Retrieve Data Sets.
+* Create Data Sets
+* Retrieve API Tokens.
+* Create API Tokens.
+
+JSON Object Queries
+-------------------
+
+JSON object queries are specified as a Java String of JSON and has a similar syntax to that of MongoDB.
+The simplest type of object query is below, which simply searches for records that have any field with term "meow":
+
+.. code-block:: json
+
+ {
+   "$any" : "meow"
+ }
+
+The client can specify this query as a Java String object and a JSON framework can be used to construct the object.
+However, if a JSON framework is not being used and you are constructing the JSON string manually, care must be taken to format the JSON correctly.
+This is especially true when writing the quote character; it must be escaped.
+
+Here is an example:
+
+.. code-block:: java
+
+ List<TDataSetResult> results = client.query(
+   "{ \"$any\" : \"meow\" }",
+   Client.DEFAULT_DATASET_NAMES,
+   Client.DEFAULT_OFFSET,
+   Client.DEFAULT_LIMIT,
+   Client.DEFAULT_FIELDS);
+
+Note that all parameters except for the query are using the defaults.
+Often, you will find that the defaults work well enough.
+However, one default you my want to change is the Data Sets to query.
+By default, all Data Sets are queried and this may not be optimal for your use case.
+The same applies for record fields; by default, all record fields are returned.
+You can specify that only certain fields are returned, this could result in higher performance if you know you do not need the whole record.
+
+Here is an example that limits the Data Sets searched and the record fields that are returned:
+
+.. code-block:: java
+
+ import java.util.Arrays;
+ ...
+ List<TDataSetResult> results = client.query(
+   "{ \"$any\" : \"meow\" }",
+   Arrays.asList("ds1", "ds2"),
+   Client.DEFAULT_OFFSET,
+   Client.DEFAULT_LIMIT,
+   Arrays.asList("score"));
+
+The object query language is very expressive as it supprts greater than, less than, and, and or operators.
```

Comment:
I would reword this.  "... as it supports or, and, greater than, and less than operators."